tests/product/image_builder.py

```diff
@@ -0,0 +1,203 @@
+from subprocess import check_output
+from sys import argv, exit
+from tests.configurable_cluster import ConfigurableCluster
+from tests.docker_cluster import DockerCluster, DockerClusterException
+from tests.no_hadoop_bare_image_provider import NoHadoopBareImageProvider
+
+from tests.product.mode_installers import StandaloneModeInstaller, \
+    YarnSliderModeInstaller
+from tests.product.prestoadmin_installer import PrestoadminInstaller
+from tests.product.standalone.presto_installer import StandalonePrestoInstaller
+from tests.product.topology_installer import TopologyInstaller
+from tests.product.yarn_slider.slider_installer import SliderInstaller
+
+import argparse
+
+
+class ProductTestImageBuilder:
```

Comment:
As part of this refactor you should edit `BaseProductTest` to use this class to generate its cluster, otherwise we end up with a lot of code duplication. There's logic to create Docker images/clusters here and in `BaseProductTest`.